### DISCUSSION TITLE
Metadata Api: Open Graph Type og:product

### DISCUSSION THREAD

**Discussion by davin-young** [vercel/next.js#48314] [Help]
### Summary

After playing around with Next Seo and realizing that didnt work for me with dynamic data, I switched my page to using the new App directory. The new generate metadata function works very well for loading the meta on page generation, however I was trying to use the opengraph type 'og:products' . https://developers.facebook.com/docs/payments/product/

This throws an error "Invalid OpenGraph type: og:product". What is the future for this?

```
export async function generateMetadata({ params }: any): Promise<Metadata> {
    const product = await fetcher(`/v1/store/product?slug=${params.productHandle}`)
    return {
        title: product.title,
        description: product.ai_data.description,

        product: { // this metadata
          price: {
            amount: product.priceData.shop_list_price,
            currency: 'USD',
          }
        },

        openGraph: {

            type: 'og:product', // This type


            url: `https://www.example.com/${ROUTES.PRODUCTS}/${product.slug}`,
            title: product.title,
            description: product.ai_data.description,
            images: product.images?.map((imageUrl: string) => ({
                url: imageUrl,
            })),
        }
    }
}
```

**Comment by AhegaoXXX** [comment]
Hello. Did you solve the problem?

**Comment by Saturate** [comment]
I think it's pretty bad to throw a 500 error on this, invalid meta data should not break the entire page IMO. It's fine to throw a warning or something.

**Comment by ogi-joo** [comment]
Any update on this?

**Reply by Saturate** [reply]
I might try to do a PR for supporting namespaced types: https://ogp.me/#types and or just throw a warning instead of an error.

**Comment by indicozy** [comment]
You can hack around it by applying it in `other` key:
```
      other: {
        ['og:type']: 'product',
      },
```

**Reply by ermakovich** [reply]
this renders `<meta name="og:type"` rather than `<meta property="og:type"` (Next 15.4.7). If you have another `{ openGraph: { type } }` defined in parent segments/layouts, it will render it alongside with this extra meta tag.

**Comment by Number16BusShelter** [comment]
This is still an issue btw

**Comment by IvanAdmaers** [comment]
Thanks to @indicozy. His solution works:

```jsx
other: {
    ['og:type']: 'product.group',
},
```

Add it to your `return` in `generateMetadata`.
